db/options.go

```diff
@@ -68,6 +69,13 @@ func WithNewToken(t thread.Token) NewOption {
 	}
 }
 
+// WithNewThreadKey provides control over thread keys to use with a db.
+func WithNewThreadKey(key thread.Key) NewOption {
```

Comment:
This was required to be able to get some tests passing again, but is also useful/required.